src/rgw/rgw_main.cc

```diff
@@ -403,10 +403,7 @@ int main(int argc, const char **argv)
 
       fe = new RGWFCGXFrontend(fcgi_pe, config);
     } else if (framework == "civetweb" || framework == "mongoose") {
-      int port;
-      config->get_val("port", 80, &port);
-
-      RGWProcessEnv env = { store, &rest, olog, port };
+      RGWProcessEnv env = { store, &rest, olog, 0 };
```

Comment:
Ok, that's related to : http://tracker.ceph.com/issues/16548 .  I sent mail 20160628 about that where I went through the code that does this.  The basic problem is that the current code path that uses that assumes that it can guess the server port based solely on whether the request came in with ssl or not.  That means not more than 1 ssl and 1 non-ssl port.  Do we really want to enforce that?